File: backend/translation/deepl/languages.go
package deepl

import (
	"encoding/json"
)

type Language struct {
	Code string `json:"language"`
	Name string `json:"name"`
}

type LanguageType string

const (
	SourceLanguages LanguageType = "source"
	TargetLanguages LanguageType = "target"
)

func (d *DeepL) GetLanguages(languageType LanguageType) ([]Language, error) {
	resp, err := d.request("/languages?type="+string(languageType), nil)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var languages []Language
	if err := json.NewDecoder(resp.Body).Decode(&languages); err != nil {
		return nil, err
	}

	return languages, nil
}

File: app/src/lib/languages/languages.ts
import type {Language} from "$lib/languages/api";

export let targetLanguages: string[] = []

export const setTargetLanguages = (languages: Language[]) => {
    targetLanguages = [...new Set(languages.map(language => language.language.split("-")[0]))]
}

export let sourceLanguageCode = ""
export const setSourceLanguageCode = (code: string) => {
    sourceLanguageCode = code
}
